CHANGES.md

```diff
@@ -1,3 +1,87 @@
+# 3.4.0 - 27 November 2025
+
+## Breaking Changes
+
+* Fix: Change JujuContext.machine_id from int to str (#2108)
```

Comment:
This was already shipped in 3.3.1.

I wonder if we need to highlight this again on the minor bump, or just drop it.

Reply:
I think we should include it to make it clear that it also made it into 3.4.0, rather than something that only went into a 3.3 branch.